app/ts/components/SendEthPage.tsx

```diff
@@ -48,40 +44,61 @@ const Main = () => {
+		event.preventDefault()
+
+		if (!accountStore.value.isConnected) {
+			accountStore.value.connectMutation.dispatch()
+			return
+		}
+
+		if (!transfer.value.isSigned) {
+			transfer.value.transactionResponseQuery.dispatch()
+			return
+		}
+
+		window.location.href = `#tx/${transfer.value.transactionResponse.hash}`
+		return
+	}
+
+	return <form onSubmit={handleSubmit}>{children}</form>
+}
+
 const SendGuide = () => {
-	const { asyncResponse } = useTransfer()
-	const [transactionResponse] = asyncResponse
+	const transferStore = useTransferStore()
 
-	switch (transactionResponse.state) {
+	if (transferStore.value.isSigned) return <Guide title='Request Successfully Sent!' content="Your transaction is awaiting confirmation from the chain. You may click on the View Transaction button to check it's status." />
+
+	const transaction = transferStore.value.transactionResponseQuery.transport
+	switch (transaction.value.state) {
 		case 'inactive':
-			return (
-				<Guide title='What happens when I click send?' content='This app will forward your request to the wallet you chose to connect with. The connected wallet handles signing and submitting your request to the chain.' />
-			)
+			return <Guide title='What happens when I click send?' content='This app will forward your request to the wallet you chose to connect with. The connected wallet handles signing and submitting your request to the chain.' />
 		case 'pending':
-			return (
-				<Guide title='Awaiting wallet confirmation...' content='At this point, your connected wallet will need action to proceed with this transaction. Carefully check the information before accepting the wallet confirmation.' />
-			)
+			return <Guide title='Awaiting wallet confirmation...' content='At this point, your connected wallet will need action to proceed with this transaction. Carefully check the information before accepting the wallet confirmation.' />
+		case 'rejected': {
+			let errorMessage = isEthereumJsonRpcError(transaction.value.error) ? transaction.value.error.error.message : transaction.value.error.message
+			return <Guide title='Wallet returned an error!' content='Check that your inputs are correct and click Send again.' quote={errorMessage} />
+		}
 		case 'resolved':
-			return (
-				<Guide title='Request Successfully Sent!' content='Your transaction is awaiting confirmation from the chain. You may click on the View Transaction button to check it&apos;s status.' />
-			)
-		case 'rejected':
-			return (
-				<Guide title='Wallet returned an error!' content='Check that your inputs are correct and click Send again.' quote={transactionResponse.error.message} />
-			)
-		default: assertUnreachable(transactionResponse)
+			return null
+		default:
+			assertUnreachable(transaction.value)
 	}
 }
 
-const Guide = ({ title, quote, content }: { title: string; quote?: string, content: string }) => {
+const Guide = ({ title, quote, content }: { title: string; quote?: string; content: string }) => {
```

Comment:
Style: Consistency.  I think we use commas everywhere else?
```suggestion
const Guide = ({ title, quote, content }: { title: string, quote?: string, content: string }) => {
```

Reply:
Somewhere else means another project? Auto-format transforms it to semi-colons just as I changed it so I doubt it's in this project. Created a separate PR if we are going to use a semicolon convention on types #66